### PR TITLE
At/cb 182 handling errors payroll discount authorization

### PR DESCRIPTION
**Pull request by Angelbt12**
handling errors: payroll-discount-authorization
f the application cannot reach the payroll discount authorization service this information should be passed to the user.

Current behavior: By deleting the payroll discount authorization mock, it looks like there aren’t documents. However, there are documents that aren’t loaded.

Expected behavior: If the service is late, the frontend application should render skeletons with the structure of messages, representing that the application is looking for information.

If the service presented an error, there should render the “information not found” component

Add to the previous activities, you should configure the behavior for the error alert as the current behavior for the trace mock

image
https://inube.atlassian.net/jira/software/projects/CB/boards/20?selectedIssue=CB-182

### CODE REVIEW

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]

File: src/components/layout/UnfoundData/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Button, Stack, Text, useMediaQuery } from "@inube/design-system";
+import { ResponsiveImage } from "./styles";
+
+interface UnfoundDataProps {
+  image?: string;
+  title: string;
+  description: string;
+  buttonDescription: string;
+  route: string;
+  onRetry?: () => void;
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/layout/UnfoundData/index.tsx:10]: ok

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx [hunk 2]
```diff
@@ -96,37 +124,55 @@ export const PromissoryNotes = (props: IPromissoryNotesProps) => {
...
+          ) : (
+            <TableBoard
+              id="promissoryNotes"
+              titles={titlesFinanacialReporting}
+              entries={dataPromissoryNotes}
+              actions={tableBoardActions}
+              actionMobile={tableBoardActionMobile}
+              loading={loading}
+              appearanceTable={{
+                widthTd: !isMobile ? "100" : "23%",
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:160]: ok

File: src/components/layout/UnfoundData/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const { image, title, description, buttonDescription, route, onRetry } =
+    props;
+  const smallScreen = useMediaQuery("(max-width: 720px)");
+
+  return (
+    <Stack
+      direction={smallScreen ? "column-reverse" : "row"}
+      gap={smallScreen ? "16px" : "8px"}
+      alignItems="center"
+      margin="s300"
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/layout/UnfoundData/index.tsx:23]: ok

File: src/components/layout/UnfoundData/styles.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import styled from "styled-components";
+
+export const ResponsiveImage = styled.img<{ $smallScreen: boolean }>`
+  display: ${({ $smallScreen }) => ($smallScreen ? "block" : "none")};
+  max-width: 100%;
+  height: auto;
+  margin-top: 8px;
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/layout/UnfoundData/styles.ts:7]: según entiendo eso era con el antiguo sistema de diseño ,ya ahora se le pase directamente

File: src/components/modals/CreditLimit/Stories/CreditLimit.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      )}
+    </>
+  );
+};
+
+Default.args = {
+  title: "Title",
+  portalId: "portal",
+  content: <Text>contenido sobre del modal</Text>,
+  buttonLabel: "Close", // Incluye el argumento buttonLabel
```
Comment (diegoAndres01) [inline src/components/modals/CreditLimit/Stories/CreditLimit.stories.tsx:32]: Remove comment

File: src/components/modals/CreditLimit/index.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+export interface IListModalProps {
+  title: string;
+  handleClose: () => void;
+  onSubmit?: () => void;
+  portalId?: string;
+  content?: JSX.Element | JSX.Element[] | string;
+  optionButtons?: IOptionButtons;
+  buttonLabel?: string; // Asegúrate de incluir buttonLabel aquí
```
Comment (diegoAndres01) [inline src/components/modals/CreditLimit/index.tsx:31]: Remove comment

File: src/components/modals/CreditLimit/index.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            </StyledSection>
+            <StyledSection>
+              <Text>Cupo disponible sin garantía</Text>
+              <StyledContainerValue $highlight>
+                $ 10'000.000
+              </StyledContainerValue>
+            </StyledSection>
+          </StyledContainerSection>
+
+          {/* Botón de actualización */}
```
Comment (diegoAndres01) [inline src/components/modals/CreditLimit/index.tsx:109]: Remove comment

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]

**Review by diegoAndres01 — APPROVED** [review]

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx [hunk 8]
```diff
@@ -72,9 +78,31 @@ export const PromissoryNotes = (props: IPromissoryNotesProps) => {
...
+      if (dataPrommisseNotes.length > 0) {
+        setDataPromissoryNotes(dataPrommisseNotes);
+        setLoading(false);
+      } else {
+        throw new Error("No se encontraron datos.");
+      }
+    } catch (err) {
+      errorObserver.notify({
+        id: "PromissoryNotes",
+        message: "Error al obtener los datos de Pagarés y Libranzas",
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:91]: It is expected that the application returns the error reported by the backend. Therefore, i expect that the message uses the "error"

File: src/components/layout/UnfoundData/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (carin91) [reply src/components/layout/UnfoundData/index.tsx:12]: Relocate the route prop to line 11

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx [hunk 9]
```diff
@@ -96,37 +124,55 @@ export const PromissoryNotes = (props: IPromissoryNotesProps) => {
...
-              title: "primary",
-              isStyleMobile: true,
-            }}
-            isFirstTable={true}
-            infoItems={infoItems}
-          />
+          {showRetry ? (
+            <UnfoundData
+              title="Error al cargar datos"
+              description="Hubo un error al intentar cargar los datos. Por favor, intente nuevamente."
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:146]: It is expected that the application returns the error reported by the backend. Therefore, i expect that the message uses the "error"

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx [hunk 10]
```diff
@@ -8,6 +8,7 @@ import { Fieldset } from "@components/data/Fieldset";
 import { TableBoard } from "@components/data/TableBoard";
 import { IEntries } from "@components/data/TableBoard/types";
 import { PromissoryNotesModal } from "@components/modals/PromissoryNotesModal";
+import { UnfoundData } from "@components/layout/UnfoundData";
```
Comment (carin91) [inline src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:12]: Adjust the imports accordingly to the standard

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx (hunk 8 above)
Comment (Angelbt12) [reply src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx]: ok

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx (hunk 9 above)
Comment (Angelbt12) [reply src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:155]: ok

File: src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx (hunk 10 above)
Comment (Angelbt12) [reply src/pages/board/outlets/financialReporting/PromissoryNotes/index.tsx:12]: ok

**Review by carin91 — APPROVED** [review]

**Pull request merged**